### ISSUE TITLE
Add support for "expandable parameters"

### ISSUE THREAD

**Issue by rvhonorato** [haddocking/benchmark-tools#46]
### Please describe the feature you'd like
Haddock3 has "expandable parameters", such as [`hisd_N`](https://github.com/haddocking/haddock3/blob/23f778c1ed94388df6788804724db170681a6cea/src/haddock/modules/topology/topoaa/defaults.yaml#L121-L130)

When the benchmark is being set, we `LoadHaddock3Params` by walking the directory structure and looking for `yaml` files;

https://github.com/haddocking/benchmark-tools/blob/990b238ed4e3bc5c737da36981d0535f6b007b05/input/input.go#L309-L349

And then for each `.yaml` file the topmost key is defined as a parameter; for example for `flexref` [this file](https://github.com/haddocking/haddock3/blob/23f778c1ed94388df6788804724db170681a6cea/src/haddock/modules/refinement/flexref/defaults.yaml) is parsed and the parameters will be

```text
mol_fix_origin_1
ambig_fname
previous_ambig
unambig_fname
...
(etc)
```

However the "expandable parameters" are not specified in this `defaults.yaml`, so the question is; how can they be identified? do they follow any sort of pattern? 

Help me out here @mgiulini 😅 

### Why is it relevant?
So that the users can access the full capabilities of HADDOCK3

**Comment by rvhonorato** [comment]
We first identified this when trying to use `resdic_N` from [`rmsdmatrix`](https://github.com/haddocking/haddock3/blob/23f778c1ed94388df6788804724db170681a6cea/src/haddock/modules/analysis/rmsdmatrix/defaults.yaml#L13-L25)

**Comment by joaomcteixeira** [comment]
Hi,

The "expandable parameters" follow a bit of intricate logic. The documentation and logic to identify and multiply expandable parameters are in this file:

https://github.com/haddocking/haddock3/blob/23f778c1ed94388df6788804724db170681a6cea/src/haddock/gear/expandable_parameters.py

This logic was defined **before** we implemented the `.yaml` files to define parameters in HADDOCK3.

Considering HADDOCK3 now uses YAMLs to define the parameters, it is very likely possible to redefine the way expandable parameters are identified and parsed. For example, declaring in the yaml if the parameter is expandable and the logic to expand it.

Best,

**Comment by joaomcteixeira** [comment]
That `gear` file reads the configuration file, identifies the expandable parameters following the four rules, and multiplies them according to the user input. the `gear.expandable_parameters` is mostly used here in the `prepare_run` file:

https://github.com/haddocking/haddock3/blob/23f778c1ed94388df6788804724db170681a6cea/src/haddock/gear/prepare_run.py#L708

The four rules were defined after discussing the topic in detail. They were all needed by the time I wrote the file, but I am not aware if some were dropped afterwards, but I think not.

**Comment by rvhonorato** [comment]
So based on what is written there any parameter written in the `defaults.yml` with the pattern `<value>_<int:i>` could be incremented up to a given `max i` and expanded to `<value>_<int:i>_<int:j>` with a separate `max j`, is that right?

If that is correct I could just get them with a regex; `(.)_\d?` -> [here](https://regex101.com/r/gHsCW1/1)

**Comment by joaomcteixeira** [comment]
It is not that straightforward. But almost. There are 4 types of expandable logics. At least you would need 4 regexes, if possible to do in such way. The logic for each group is defined in the file. For example, the first group logic is defined here:

https://github.com/haddocking/haddock3/blob/23f778c1ed94388df6788804724db170681a6cea/src/haddock/gear/expandable_parameters.py#L482

The `prepare_run` uses the logics [here](https://github.com/haddocking/haddock3/blob/23f778c1ed94388df6788804724db170681a6cea/src/haddock/gear/prepare_run.py#L709-L722).

The order in which the logics are identified matters (if I remember well).

**Comment by rvhonorato** [comment]
Indeed, but here we don't need to replicate the whole logic here. The point of this part of the input validation of the benchmark is to "fail fast", so if you define a parameter that does not exist it exits with an error. Whatever is defined in the scenario is passed to the `run.toml` regardless - so if someone defines an expandable parameter (`<value>_<int:i>_<int:j>`) with the wrong `i` and `j`, haddock3 will still catch it. 

Currently it will only acept things that are defined in the `defaults.yml` explicitly, so the regex would be sufficient (for now I think) - it could work like this:

```mermaid
graph TD
    A[read default.yml] --> B{perfect match}
    B --> D[yes]
    B --> E[no]
    D --> F[pass]
    E -->|regex| G{is expandable}
    G --> H[yes]
    H --> F
    G --> I[no]
    I --> j[exit]
```

**Comment by joaomcteixeira** [comment]
Okay. Have a look to the different logics then to transfer them to regex properly. Otherwise, if all this is in python, you can directly use the functions indicated in the links above.

Please also note this -> https://github.com/haddocking/haddock3/blob/23f778c1ed94388df6788804724db170681a6cea/src/haddock/gear/expandable_parameters.py#L103-L113
